### ISSUE TITLE
Setup module reports facter_virtual as physical on OS KVM instance

### ISSUE THREAD

**Issue by jkomara** [ansible/ansible#37652]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
setup

##### ANSIBLE VERSION

```
ansible 2.4.3.0
ansible 2.6.0
```

##### CONFIGURATION

```
ANSIBLE_SSH_ARGS = -o UserKnownHostsFile=/dev/null -o ServerAliveInterval=60 -o ServerAliveCountMax=60
ANSIBLE_SSH_CONTROL_PATH = %(directory)s/%%h-%%r
DEFAULT_SCP_IF_SSH = True
```
##### OS / ENVIRONMENT

Host
```
$ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 10.13.3 (17D102)
      Kernel Version: Darwin 17.4.0
```
Target System
```
$ cat /etc/redhat-release
CentOS Linux release 7.4.1708 (Core)

$ dmidecode -t1
System Information
	Manufacturer: Fedora Project
	Product Name: OpenStack Nova

$ facter -v
3.6.2
```

##### SUMMARY

The setup module is incorrectly reporting that an Openstack instance's `virtual` facter fact is physical.

##### STEPS TO REPRODUCE

```yaml
$ ansible -i test-os-1, all -m setup -a "filter=facter_virtual"
```

##### EXPECTED RESULTS

When running the steps above I would expect that the setup module would return the same value that facter is returning from the target host.

```
test-os-1$ facter virtual
kvm
```

##### ACTUAL RESULTS

The setup module returns the `facter_virtual` fact set to physical.

```
test-os-1 | SUCCESS => {
    "ansible_facts": {
        "facter_virtual": "physical"
    },
    "changed": false
}
```
##### NOTE
The `ansible_virtualization_role` is correctly reporting that the target system is a "guest" so I am able to key off of that instead.
```
test-os-1 | SUCCESS => {
    "ansible_facts": {
        "ansible_virtualization_role": "guest",
        "ansible_virtualization_type": "openstack"
    },
    "changed": false
}
```

I did find that this has been reported in the past but it looks like it was not able to be replicated: https://github.com/ansible/ansible/issues/15165

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/system/setup.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/setup.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gl0bus** [comment]
As far as I can see, the facts with the `facter_` prefix are delivered by facter (as long as installed),
so the related issue should probably be opened in https://github.com/puppetlabs/facter

**Comment by Akasurde** [comment]
@jkomara Could you please paste output of following command - 

```
# facter --json
```
@gl0bus is right. `facter` command may be reporting wrong facts.

**Comment by Akasurde** [comment]
I see one resolved issue - https://tickets.puppetlabs.com/browse/FACT-1043

**Comment by jkomara** [comment]
The output in the 'expected results' section was the output of facter on the box. It was reporting 'kvm' for the virtual fact even though ansible was returning `"facter_virtual": "physical"`.

I went out to get you a fresh example and now I am unable to reproduce the problem. I am not sure what has changed from then to now. I appreciate you following up. If I can figure out what changed or what I did wrong I will make sure to post back on this issue.

Thank you.
